### PR TITLE
[test]:improve coverage of for application.go in pkg/metaserver

### PR DESCRIPTION
**Pull request by Dhiren-Mhatre**
/kind test

**What this PR does / why we need it**:
This PR enhances test coverage for the metaserver application.go file as part of the [lfx-mentorship] program. The changes increase code coverage from 55% to 95% by:

1. Adding tests for previously uncovered methods like ToBytes(), Cancel(), Reset(), Add(), Close(), LastCloseTime(), and Wait()
2. Improving existing tests with additional test cases and edge case handling
3. Adding error path testing for JSON unmarshaling operations

**Which issue(s) this PR fixes**:
Part of [lfx-mentorship] Enhance KubeEdge testing coverage #6101

**Special notes for your reviewer**:
This PR only modifies the test file (application_test.go) and doesn't change any functional code. All new tests pass successfully.

**Does this PR introduce a user-facing change?**:
```
NONE
```

### CODE REVIEW

File: pkg/metaserver/application_test.go [hunk 1]
```diff
@@ -424,3 +469,160 @@ func TestMsgToApplications(t *testing.T) {
...
+	select {
+	case <-ctx.Done():
+	default:
+		t.Error("Context was not canceled")
+	}
+
+	app = Application{
+		cancel: nil,
+	}
+	app.Cancel()
```
Comment (wbc6080) [inline pkg/metaserver/application_test.go:521]: Are there two situations that consider whether cancel is empty? You can add some comments to make the code easier to understand

File: pkg/metaserver/application_test.go [hunk 2]
```diff
@@ -424,3 +469,160 @@ func TestMsgToApplications(t *testing.T) {
...
+	select {
+	case <-ctx.Done():
+	default:
+		t.Error("Context should be done")
+	}
+
+	app = Application{
+		ctx: nil,
+	}
+	app.Wait()
```
Comment (wbc6080) [inline pkg/metaserver/application_test.go:627]: ditto

File: pkg/metaserver/application_test.go (hunk 1 above)
Comment (Dhiren-Mhatre) [reply pkg/metaserver/application_test.go]: The first test case verifies normal cancellation works when cancel function is provided
 and the second test case verifies defensive programming that prevents panics when cancel is nil. I've updated the code with clear comments to distinguish between the test scenarios

**Comment by wbc6080** [comment]
lgtm
cc @WillardHu @fisherxu @Shelley-BaoYue

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve
